car_dreamer/carla_follow_env.py

```diff
@@ -0,0 +1,188 @@
+import carla
+import math
+import numpy as np
+import random
+
+from .carla_wpt_env import CarlaWptEnv
+from .toolkit import FixedEndingPlanner, get_vehicle_pos, get_location_distance, get_vehicle_velocity, get_vehicle_orientation
+
+class CarlaFollowEnv(CarlaWptEnv):
+    """
+    Vehicle follows the car in front of it at a reasonable distance for going straight, turning left, and turning right.
+
+    **Provided Tasks**: ``carla_follow``
+    """
+
+    def on_reset(self) -> None:
+        # checks which path the vehicle will take this time
+        max_num_of_directions = len(self._config.lane_start_points)
+        if self._config.direction == max_num_of_directions:
+            self.random_num = random.randint(0, max_num_of_directions - 1)    # random path
+        elif self._config.direction >= 0 and self._config.direction < max_num_of_directions:
```

Comment:
it seems that each direction values corresponds to specific meanings. it would be more readable to have enum constants defined in the utils